tests/unit/IO/test_vt_writer.py

```diff
@@ -38,25 +38,35 @@ def __list_optional_keys_recursive(self, nodes: Any, dot_path: str = ''):
             return opt_nodes
 
         # Key & node
-        if isinstance(nodes, dict):
+        elif isinstance(nodes, dict):
```

Comment:
These nested conditionals are hard to follow/understand. Please comment this code to express intent

Reply:
Done